src/libraries/System.Private.CoreLib/src/System/Reflection/MethodBase.cs

```diff
@@ -282,31 +297,25 @@ BindingFlags invokeAttr
         [StructLayout(LayoutKind.Sequential)]
         private protected ref struct StackAllocedArguments
         {
-            internal object? _arg0;
-#pragma warning disable CA1823, CS0169, IDE0051, IDE0044 // accessed via 'CheckArguments' ref arithmetic
-            private object? _arg1;
-            private object? _arg2;
-            private object? _arg3;
-#pragma warning restore CA1823, CS0169, IDE0051, IDE0044
-            internal ParameterCopyBackAction _copyBack0;
-#pragma warning disable CA1823, CS0169, IDE0051, IDE0044 // accessed via 'CheckArguments' ref arithmetic
-            private ParameterCopyBackAction _copyBack1;
-            private ParameterCopyBackAction _copyBack2;
-            private ParameterCopyBackAction _copyBack3;
-#pragma warning restore CA1823, CS0169, IDE0051, IDE0044
+            internal ArgumentData<object?> _arg0;
+            internal ArgumentData<ParameterCopyBackAction> _copyBack0;
```

Comment:
```suggestion
            internal ArgumentData<object?> _arg;
            internal ArgumentData<ParameterCopyBackAction> _copyBack;
```
`0` does not make sense in these field names.

Reply:
Right, I did not want to change the consuming code much, but removing 0 makes sense and not too big of a change.

Reply:
Since I need to change consuming code for this, I will add `AsSpan()` helper as well.